bok_choy/page_object.py

```diff
@@ -330,6 +345,25 @@ def _verify_page(self):
             )
             raise WrongPageError(msg)
 
+    def _verify_xss_exposure(self):
+        """
+        Verify that there are no obvious XSS exposures on the page (based on test authors
+        including XSS_INJECTION in content rendered on the page).
+
+        If an xss issue is found, raise a 'XSSExposureError'.
+        """
+        # Use innerHTML to get dynamically injected HTML as well as server-side HTML.
+        html_source = self.browser.execute_script(
+            "return document.documentElement.innerHTML.toLowerCase()"
+        )
+
+        # Check taken from https://www.owasp.org/index.php/XSS_Filter_Evasion_Cheat_Sheet#XSS_Locator.
+        all_hits = html_source.count('<xss')
+        if all_hits > 0:
+            value_hits = len(re.findall(r'value="\'\';!--.*<xss.*{\(\)}"', html_source))
+            if all_hits > value_hits:
+                raise XSSExposureError("{} XSS issue(s) found on page".format(all_hits - value_hits))
```

Comment:
would it be good to have a debug log with the html source, since it's created & lower-cased above? Just thinking of helping troubleshoot if an issue is found.